src/NodeApi/Interop/JSSynchronizationContext.cs

```diff
@@ -7,6 +7,25 @@
 
 namespace Microsoft.JavaScript.NodeApi.Interop;
 
+/// <summary>
+/// Manages the synchronization context for a JavaScript environment, allowing callbacks and
+/// asynchronous continuations to be invoked on the JavaScript thread that runs the environment.
+/// </summary>
+/// <remarks>
+/// All JavaScript values are bound to the thread that runs the JS environment and can only be
+/// accessed from the same thread. Attempts to access a JavaScript value from a different thread
+/// will throw <see cref="JSInvalidScopeException" />.
```

Comment:
Can we have a better named exception in such case? E.g. `JSInvalidThreadException`.

Reply:
Renamed.